robot_sort/robot_sort.py

```diff
@@ -97,13 +97,60 @@ def sort(self):
         Sort the robot's list.
         """
         # Fill this out
-        pass
 
+        while True:
+            # grab first item
+            if self.compare_item() == None:
+                self.swap_item()
+            self.set_light_off()
+            while self.can_move_right() == True:
+
+                self.move_right()
+                if self.compare_item() == 1:
+                    # print('Swapping items...')
+                    self.swap_item()
+                    self.move_left()
+                    self.swap_item()
+                    self.move_right()
+                    self.set_light_on()
+                # if next item is smaller than held item, move backwards 1 and put item back
+                if self.compare_item() == 0 or self.compare_item() == -1:
+                    # print('Swapping items...')
+                    self.move_left()
+                    self.swap_item()
+                    self.move_right()
+                    self.swap_item()
+                # Put largest int at end of list
+                if self.can_move_right() == False and self._item != None:
+                    # print('Swapping items...')
+                    self.swap_item()
+            # Move to beginning of list
+            while self._position > 0:
+                self.move_left()
+            # No swaps, list is sorted
+            if self.light_is_on() == False:
+                print('Count: ', self._time)
+                break
+
```

Comment:
very nice attempt; code could be cleaned up a bit to adhere to the DRY principle